### ISSUE TITLE
Add configuration for location vagrant file

### ISSUE THREAD

**Issue by lykciv** [phpro/grumphp#153]
| Q | A |
| --- | --- |
| Branch | master |
| Bug? | no |
| New feature? | yes |
| Question? | yes |
| Documentation? | no |

I'm trying out the new setup with the vagrant hooks preset.

My project structure is as follows:
- src
  - .git
  - vagrant
    - Vagrantfile

The vagrant hooks will run the vagrant commands from the root of the project, which means that I'm getting this error:

```
A Vagrant environment or target machine is required to run this
command. Run `vagrant init` to create a new Vagrant environment. Or,
get an ID of a target machine from `vagrant global-status` to run
this command on. A final option is to change to a directory with a
Vagrantfile and to try again.
```

If I update the hook to first change to the vagrant directory, the hook works as expected.

```
# Fetch the GIT diff and format it as command input:
DIFF=$(git diff -r -p -m -M --full-index --staged | cat)

cd vagrant;

# Run GrumPHP
vagrant ssh --command '$(which sh)' << COMMANDS
# ...
```

Of course, I could create my own hook templates, but I think it would be useful to make this configurable?

There could be an extra parameter `vagrant_dir` with default value `./` which allows users to set the location to their Vagrantfile. In my case this would be `vagrant`.

Would this be a useful feature? If yes, I'm willing to put together a pull request for it.

Thank you!

**Comment by veewee** [comment]
Yesterday we had a similar request for additional git hook parameters in the YAML file.
(see https://github.com/phpro/grumphp/issues/152 for more information).

With the vagrant preset files, we tried to avoid setting additional parameters in the yaml file. We did this to make sure that the configuration was preset independent. The disadvantage of this approuch is that a home directory, ... needs to be set in the vagrant box.

Maybe it's a better idea to provide something like a git_hooks_variables parameter which can be used to fill replacements. For example:

```
parameters:
    git_hook_variables:
         VAGRANT_LOCAL_PATH: ./
         VAGRANT_REMOTE_PATH: /vagrant
```

With this approach the git:init command can replace the variables in the git hooks. Maybe it's best to make the replacements bash variables: ${VAGRANT_LOCAL_PATH} to make sure that code inspection doens't flip.

I'm not sure if @jverdeyen is going to work on this part as mentioned in ticket #152. If not: feel free to implement! :)
